src/machines/java/com/enderio/machines/common/blockentity/XPObeliskBlockEntity.java

```diff
@@ -0,0 +1,85 @@
+package com.enderio.machines.common.blockentity;
+
+import com.enderio.base.common.init.EIOFluids;
+import com.enderio.base.common.tag.EIOTags;
+import com.enderio.base.common.util.ExperienceUtil;
+import com.enderio.core.common.network.slot.IntegerNetworkDataSlot;
+import com.enderio.machines.common.blockentity.base.MachineBlockEntity;
+import com.enderio.machines.common.menu.XPObeliskMenu;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.templates.FluidTank;
+import org.jetbrains.annotations.Nullable;
+
+public class XPObeliskBlockEntity extends MachineBlockEntity {
+
+    IntegerNetworkDataSlot xpTankDataSlot;
+
+    public XPObeliskBlockEntity(BlockEntityType<?> type, BlockPos worldPosition, BlockState blockState) {
+        super(type, worldPosition, blockState);
+
+        this.xpTankDataSlot = new IntegerNetworkDataSlot(() -> getFluidTankNN().getFluidAmount(),
+            amount -> getFluidTankNN().setFluid(new FluidStack(EIOFluids.XP_JUICE.getSource(), amount)));
+        addDataSlot(xpTankDataSlot);
+    }
+
+    @Nullable
+    @Override
+    public AbstractContainerMenu createMenu(int containerId, Inventory playerInventory, Player player) {
+        return new XPObeliskMenu(this, playerInventory, containerId);
+    }
+
+    @Override
+    protected @Nullable FluidTank createFluidTank() {
+        // What happens when some other fluid with same tag is inserted ?
```

Comment:
From my understanding it will match based on current fluid too - if you want it to accept XP indiscriminately, check the soul binder conversion logic :)